sql/updates/latest-dev.sql

```diff
@@ -118,3 +118,80 @@ ALTER TABLE IF EXISTS _timescaledb_catalog.compression_chunk_size ADD COLUMN IF
+--       constraint.
+--    2. Parse the constraint expression and extract the column name,
+--       and upper and lower range values as text.
+--    3. Use the column type to construct the range values (UNIX
+--       microseconds) from these values.
+
+WITH
+   -- All dimension slices that are mentioned in the chunk_constraint
+   -- table but are missing from the dimension_slice table.
+   missing_slices AS (
+      SELECT DISTINCT dimension_slice_id
+		    , constraint_name
+		    , attname AS column_name
+		    , pg_get_expr(conbin,conrelid) AS constraint_expr
+	     FROM _timescaledb_catalog.chunk_constraint
+	LEFT JOIN _timescaledb_catalog.dimension_slice sl
+	       ON dimension_slice_id = sl.id
+	     JOIN pg_constraint ON conname = constraint_name
+	     JOIN pg_attribute ON attnum = conkey[1] AND attrelid = conrelid
+	   WHERE dimension_slice_id IS NOT NULL AND sl.id IS NULL
+   ),
+
+  -- Unparsed range start and end for each dimension slice id that
+  -- is missing.
+   unparsed_missing_slices AS (
+      SELECT dimension_slice_id
+	   , constraint_name
+	   , column_name
+	   , (SELECT SUBSTRING(constraint_expr, $$>=\s*'?([\d\s:+-]+)'?$$)) AS range_start
+	   , (SELECT SUBSTRING(constraint_expr, $$<\s*'?([\d\s:+-]+)'?$$)) AS range_end
+	FROM missing_slices
+   )
+SELECT dimension_slice_id
+     , di.id AS dimension_id
+     , CASE
+       WHEN di.column_type IN ('smallint'::regtype, 'bigint'::regtype, 'integer'::regtype) THEN
+       	    CASE
+	    WHEN range_start IS NULL
+	    THEN -9223372036854775808
+	    ELSE _timescaledb_internal.time_to_internal(range_start::bigint)
+	    END
+       WHEN di.column_type = 'timestamptz'::regtype THEN
+       	    _timescaledb_internal.time_to_internal(range_start::timestamptz)
+       WHEN di.column_type = 'timestamp'::regtype THEN
+       	    _timescaledb_internal.time_to_internal(range_start::timestamp)
+       WHEN di.column_type = 'date'::regtype THEN
+       	    _timescaledb_internal.time_to_internal(range_start::date)
+       ELSE
+	    NULL
+       END AS range_start
+     , CASE 
+       WHEN di.column_type IN ('smallint'::regtype, 'bigint'::regtype, 'integer'::regtype) THEN
+       	    CASE WHEN range_end IS NULL
+	    THEN 9223372036854775807
+	    ELSE _timescaledb_internal.time_to_internal(range_end::bigint)
+	    END
+       WHEN di.column_type = 'timestamptz'::regtype THEN
+       	    _timescaledb_internal.time_to_internal(range_end::timestamptz)
+       WHEN di.column_type = 'timestamp'::regtype THEN
+       	    _timescaledb_internal.time_to_internal(range_end::timestamp)
+       WHEN di.column_type = 'date'::regtype THEN
+       	    _timescaledb_internal.time_to_internal(range_end::date)
+       ELSE NULL
+       END AS range_end
+  FROM unparsed_missing_slices
+  JOIN _timescaledb_catalog.dimension di
+ USING (column_name);
+
+INSERT INTO _timescaledb_catalog.dimension_slice
+SELECT * FROM rebuilt_missing_slices;
```

Comment:
This doesn't seem right, `rebuilt_missing_slices` doesn't seem to exist anymore and this is the only reference to it apart from the `DROP VIEW` below.